custom_tables.md

```diff
@@ -0,0 +1,140 @@
+                "nickname": "Fredu",
+                "lucky_number": 13,
+                "favourite_ide_cd": "VIM"
+            },
+            "person_hobbies_n": [
+                {
+                    "id": 1,
+                    "name": "Knitting"
+                },
+                {
+                    "name": "Running"
+                },
+                {
+                    "id": 3,
+                    "_delete": true
+                }
+            ]
+        }
+    }
+}
+```
+
+* will change the first name of the person with reference `123456` to "Fred" (standard behaviour of `Masterdata.Person.Update`)
+* will manage the contents of the custom tables as specified (additional custom table data behaviour)
+  * the FK `person_id` of the rows custom tables will automatically be set to the id of the person with reference `123456`
+
+### Supported messages
+
+Custom data can be sent alongside any message/event type that is clearly related to a business entity. For now we save us the trouble to maintain an exhaustive list of supported messages. The system actively rejects messages and tells you via error if custom table data cannnot be sent alongside your message type.
+
+### Update behaviour
+
+For 1:1 tables
+
+* If no record is present it will be created.
+* If there already is a record, it will be updated with the fields you provide.
+* To delete an existing record, pass `{ "_delete": true }`
+  * This operation is idempotent, it will not fail if no record is present.
+  * `_delete` is not supported as a table column name.
+
+For 1:n tables, each array entry is handled as follows
+
+* If no id is present the record will be created.
+* If an id is present, the record will be updated with the fields you provide.
+* If only the keys `id` and `_delete` are present and `_delete` is `true`, the record with the given id will be deleted.
+  * `_delete` is not supported as a table column name.
+
+## Mapping of references
+
+Custom tables belong to a business entity. Furthermore, they might have FKs pointing to the ids of other business entities.
+On our APIs, we use the reference of business entities as their identifier. To allow you to fill FKs in custom tables if you only know the reference and not the id, the API
+will map given references to ids.
+
+So, given this 1:1 custom table `person_objekt_rating`
+
+| Column    | Descripton                      |
+| --------- | ------------------------------- |
+| id        | Integer, PK                     |
+| person_id | Integer, FK on table `personen` |
+| objekt_id | Integer, FK on table `objekte`  |
+| rating    | Integer                         |
+
+and an entry in `objekte` with reference `"000123.45.000678"` and id `23`,
+you can manage the table via the API as follows
+
+```json
+{
+    "data": {
+        "customTables": {
+            "person_objekt_rating": {
+                "rating": 5,
+                "objekt_reference": "000123.45.000678"
```

Comment:
will there be an error if referenz is not found?

Reply:
also, what is the person_id in this example?

Reply:
there will be an error on invalid references.
what the person_id will be is explained above. i will add eventType and personReference fields here for clarification.